WIP : Feature : Django 2.2 support and generic frame for multiple office

jbgury opened the pull request:
* Add support for django 2.2 and python 3.8
* Allow to support a generic frame for multiple office into the instance

Commit:
Migrate code and tests to python3.8/Django2.1

Commit:
Migrate code and tests to python3.8/Django2.1

Commit:
Feature: add generic frame for multiple office for the same instance. With Django 2.2

Commit:
Feature: add generic frame for multiple office for the same instance. With Django 2.2

Commit:
Feature : add support for python 3.8

Commit:
Feature : add support for python 3.8

JocelynDelalande commented on .travis.yml:
```diff
@@ -13,7 +13,7 @@ install:
   - export PATH=$PATH:$PWD
   - python ./manage.py collectstatic --no-input
 python:
-  - "3.6"
```
Are we still targetting Python 3.5 as the README states ? If so, and if 3.8 is not our minimal target, I suggest that we test with travis either :

- only the lowest version we support (3.5 ?)
- all the supported python versions

This will IMHO provide better feedback about compatibility breakage, as forward compatibility is less frequent than backward compat.

jbgury replied:
You're right, I change to add support on different version from 3.5

jbgury replied:
Support of Python 3.5 is possible, but not testable on CI. In order to trace output of requests into the robot framework, RequestsLogger has to be used, and this script is not more compatible with python 3.5 (see https://github.com/peterservice-rnd/robotframework-requestslogger/issues/5 )

littlejo replied:
End of security support of Python 3.5 is 2020-09-13. So I think minimal version should be 3.6.

jbgury replied:
See #234
I separated technical upgrade and fixes in a dedicated PR.

JocelynDelalande commented on libreosteoweb/models.py:
```diff
@@ -418,10 +431,6 @@ class OfficeSettings(models.Model):
                                               blank=True)
 
     def save(self, *args, **kwargs):
-        """
```
cleanup : overriden save method can be removed.

jbgury replied:
done... a mistake

JocelynDelalande reviewed (commented):
Thanks for that maintenance+feature update ! Won't have the time soon to test, sadly, so I just got it a quick review. Hope that helps a bit.

littlejo committed:
Timeout quand je lance yarn (#229)

* Fix timeout

* Change https => git+https

Co-authored-by: Joseph Ligier <joseph.ligier@gmail.com>

JocelynDelalande committed:
Corrige le format des .po (#218)

* Fix .po removing line numbers

As mentioned in CONTRIBUTING.md

* Check that no translation string is missing nor forgotten in CI

* Check that no migration has been forgotten in CI

Bonus !

* Fix po

To get reproducable .po content among devs

* Comment CI stages

* Add some echo-comments in CI to ease comprehension of output

jbgury committed:
Fix dev dependency, yarn issue with momentjs (#232)

* Fix CI Issue and JS dependencies (#233, #230)

Commit:
Validate against multiple python version

Commit:
Fix typo in translation key

Commit:
Fix missing translation on the multiple office frame

Commit:
Fix translation on check_sum

Commit:
Merge branch 'develop-django2' of github.com:libreosteo/Libreosteo into develop-django2

Commit:
Fix merge issue on models.py

Commit:
Add default url for multiple office frame

Commit:
Fix unit test due to management of multiple office settings

Commit:
Add prefix on invoice for multiple office
